src/plugins/intel_gpu/src/kernel_selector/kernels/non_zero/count_nonzero_kernel_ref.cpp

```diff
@@ -58,8 +56,8 @@ CommonDispatchData CountNonzeroKernelRef::SetDefault(const count_nonzero_params&
                        {Tensor::DataChannelName::FEATURE, Tensor::DataChannelName::BATCH}};
     }
 
-    dispatchData.lws =
-        GetOptimalLocalWorkGroupSizes(dispatchData.gws, params.engineInfo, in_layout, out_layout, dims_by_gws);
+    // Set 1 work group to avoid synchornization issue for summation of nonzero counting.
+    dispatchData.lws = dispatchData.gws;
```

Comment:
What if we exceed # available work group size? 
Should be changed in a way like : 
- Divide data by params.engineInfo.maxWorkGroupSize
- e.g., if max # of wgs is 512 and the total # of data is 1024, each work item should handle 2 data. 
-

Reply:
Thank you for your comments. I will update it.

Reply:
Update unit tests

Reply:
Update kernel. please review it again